src/pkg/podman/podman.go

```diff
@@ -33,6 +35,13 @@ var (
 
 var (
 	LogLevel = logrus.ErrorLevel
+	ShowLogs = false
+)
+
+var (
+	ErrUnauthorized    = errors.New("not authorized to access resources in registry")
+	ErrManifestUnknown = errors.New("registry does not know the requested manifest")
+	ErrNameUnknown     = errors.New("registry does not know the requested repository name")
```

Comment:
Umm... no. :)

Scraping the `stderr` of `podman` invocations to reconstruct the errors is a stop-gap measure for implementing authentication, and that's a very nice way to put it. Currently, I am not sure if even that much is acceptable as an upstream change.

Redirecting the `stderr` to a non-terminal file descriptor disables Podman's progress bars when using `--verbose`. The `stderr` parsing logic in `parseErrorMsg` gets more and more fragile as we try to cover more and more scenarios with it, and it will all collapse as soon as something gets tweaked in the many layers of code lying underneath `podman`.

So, no, just no. :) Let's not go overboard and try to do more of this, for some cosmetic niceties. We really should be pushing for proper exit codes, instead of going down this route.